cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -368,11 +379,29 @@ func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
 		defer cli.Close()
 
 		ctx, cancel := context.WithTimeout(context.Background(), etcdTimeout)
+		defer cancel()
+		var resp *clientv3.MemberAddResponse
 		resp, err = cli.MemberAdd(ctx, []string{peerAddrs})
-		cancel()
 		if err == nil {
+			respMembers = resp.Members
+
```

Comment:
[1] we can remove some of these new lines (compact).

Reply:
Fixed.